test/parallel_api/numeric/numeric.ops/inclusive_scan_by_segment_zip.pass.cpp

```diff
@@ -0,0 +1,112 @@
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// This file incorporates work covered by the following copyright and permission
+// notice:
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+
+#include "support/test_config.h"
+
+#include _PSTL_TEST_HEADER(execution)
+
+#include "support/utils.h"
+
+#include _PSTL_TEST_HEADER(algorithm)
+#include _PSTL_TEST_HEADER(iterator)
+#include _PSTL_TEST_HEADER(functional)
+
+#include <functional>
+#include <iostream>
+
+#if TEST_DPCPP_BACKEND_PRESENT
+#include "support/sycl_alloc_utils.h"
+
+template <sycl::usm::alloc alloc_type, typename KernelName>
+void
+test_with_usm()
+{
+    sycl::queue q = TestUtils::get_test_queue();
+
+    constexpr int n = 9;
+
+    //data initialization
+    int keys1  [n] = { 11, 11, 21, 20, 21, 21, 21, 37, 37 };
+    int keys2  [n] = { 11, 11, 20, 20, 20, 21, 21, 37, 37 };
+    int values1[n] = {  0,  1,  2,  3,  4,  5,  6,  7,  8 };
+    int values2[n] = {  0,  1,  2,  3,  4,  5,  6,  7,  8 };
+    int output_values1[n] = { };
+    int output_values2[n] = { };
+
+    // allocate USM memory and copying data to USM shared/device memory
+    TestUtils::usm_data_transfer<alloc_type, int> dt_helper1(q, std::begin(keys1),          std::end(keys1));
+    TestUtils::usm_data_transfer<alloc_type, int> dt_helper2(q, std::begin(keys2),          std::end(keys2));
+    TestUtils::usm_data_transfer<alloc_type, int> dt_helper3(q, std::begin(values1),        std::end(values1));
+    TestUtils::usm_data_transfer<alloc_type, int> dt_helper4(q, std::begin(values2),        std::end(values2));
+    TestUtils::usm_data_transfer<alloc_type, int> dt_helper5(q, std::begin(output_values1), std::end(output_values1));
+    TestUtils::usm_data_transfer<alloc_type, int> dt_helper6(q, std::begin(output_values2), std::end(output_values2));
+    auto d_keys1          = dt_helper1.get_data();
+    auto d_keys2          = dt_helper2.get_data();
+    auto d_values1        = dt_helper3.get_data();
+    auto d_values2        = dt_helper4.get_data();
+    auto d_output_values1 = dt_helper5.get_data();
+    auto d_output_values2 = dt_helper6.get_data();
+
+    //make zip iterators
+    auto begin_keys_in = oneapi::dpl::make_zip_iterator(d_keys1, d_keys2);
+    auto end_keys_in   = oneapi::dpl::make_zip_iterator(d_keys1 + n, d_keys2 + n);
+    auto begin_vals_in = oneapi::dpl::make_zip_iterator(d_values1, d_values2);
+    auto begin_vals_out= oneapi::dpl::make_zip_iterator(d_output_values1, d_output_values2);
+
+    //run inclusive_scan_by_segment algorithm 
+    oneapi::dpl::inclusive_scan_by_segment(
+        TestUtils::make_device_policy<KernelName>(q), begin_keys_in,
+        end_keys_in, begin_vals_in, begin_vals_out,
+        ::std::equal_to<>(), TestUtils::tuple_add_fn());
+
+    q.wait();
```

Comment:
Is this call really required here?

Reply:
No, I have removed it from all locations.